### PR TITLE
Make use of a better python module for gitlab_project

### PR DESCRIPTION
**Pull request by dj-wasabi**
##### SUMMARY
Using a python module that is not deprecated anymore.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
gitlab_project

##### ANSIBLE VERSION

```
ansible 2.5.2
  config file = None
  configured module search path = [u'/Users/wdijkerman/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/wdijkerman/git/ansible/environment/.virtualenv/environment/lib/python2.7/site-packages/ansible
  executable location = /Users/wdijkerman/git/ansible/environment/.virtualenv/environment/bin/ansible
  python version = 2.7.10 (default, Oct  6 2017, 22:29:07) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by dj-wasabi** [comment]
Tests is failing, even on "devel" so don't know how to proceed with this.

**Comment by Akasurde** [comment]
```
ERROR: Found 1 validate-modules issue(s) which need to be resolved:
ERROR: test/sanity/validate-modules/ignore.txt:1127:1: A102 Remove since "lib/ansible/modules/source_control/gitlab_project.py" passes "E326" test (75%)
```
This is shippable reported error. You need to remove gitlab_project from [test/sanity/validate-modules/ignore.txt at line 1127](https://github.com/ansible/ansible/blob/bf5cd98a2bba05228e9c81d16f74f91ac45efe27/test/sanity/validate-modules/ignore.txt#L1127) as it passed E326 test.

**Comment by dj-wasabi** [comment]
Hi @Akasurde 

I removed the line in the file you mentioned, but the test is still failing?

**Comment by Akasurde** [comment]
@dj-wasabi Failed testcase is related to gitlab_project - https://github.com/ansible/ansible/blob/devel/test/units/modules/source_control/test_gitlab_project.py#L58 

You can check it here - https://app.shippable.com/github/ansible/ansible/runs/67593/3/console

**Comment by dj-wasabi** [comment]
Hi @Akasurde I know. But like I said before, the test is also failing on devel so how am I get this to work?

**Comment by mattclay** [comment]
@dj-wasabi The gitlab_project unit tests are currently passing in devel. You can run them locally from your branch with:

`ansible-test units test/units/modules/source_control/test_gitlab_project.py --docker -v`

The changes in this PR are causing unit test failures:

https://app.shippable.com/github/ansible/ansible/runs/67593/3/tests

If you're unable to reproduce the test results locally, perhaps you're using a different version of `python-gitlab` than is in our docker container used for CI?

File: lib/ansible/modules/source_control/gitlab_project.py [hunk 1]
```diff
@@ -57,93 +67,129 @@
...
-            - Possible values are true and false.
+        required: false
+        type: bool
+        default: true
+    builds_enabled:
+        description:
+            - If a build creation for this project should be available or not.
+        type: bool
+        default: false
+        version_added: "2.7"
```
Comment (gundalow) [inline lib/ansible/modules/source_control/gitlab_project.py:100]: `"2.8"`

File: lib/ansible/modules/source_control/gitlab_project.py [hunk 2]
```diff
@@ -57,93 +67,129 @@
...
+    public_builds:
+        description:
+            - If true, builds can be viewed by non-project-members.
+            - Will only work if "builds_enabled" is set to True.
+        type: bool
+        default: false
+        version_added: "2.7"
```
Comment (gundalow) [inline lib/ansible/modules/source_control/gitlab_project.py:107]: `"2.8"`

File: lib/ansible/modules/source_control/gitlab_project.py [hunk 3]
```diff
@@ -57,93 +67,129 @@
...
+    only_allow_merge_if_build_succeeds:
+        description:
+            - Set whether merge requests can only be merged with successful builds.
+            - Will only work if "builds_enabled" is set to True.
         type: bool
-        default: 'yes'
+        default: false
+        version_added: "2.7"
```
Comment (gundalow) [inline lib/ansible/modules/source_control/gitlab_project.py:114]: `"2.8"`

File: lib/ansible/modules/source_control/gitlab_project.py [hunk 4]
```diff
@@ -57,93 +67,129 @@
...
+    container_registry_enabled:
+        description:
+            - Enable container registry for this project.
+            - Will only work if "builds_enabled" is set to True.
+        type: bool
+        default: false
+        version_added: "2.7"
```
Comment (gundalow) [inline lib/ansible/modules/source_control/gitlab_project.py:121]: `"2.8"`

**Comment by Lunik** [comment]
Will be fixed with #50442

**Comment by Shaps** [comment]
@dj-wasabi I believe this can be closed since @Lunik has updated most of the modules for this.

**Comment by dagwieers** [comment]
resolved_by_pr #51141

**Pull request closed without merging**